src/metabase/driver/sql/query_processor/deprecated.clj

```diff
@@ -0,0 +1,128 @@
+(ns metabase.driver.sql.query-processor.deprecated
+  "Deprecated stuff that used to live in [[metabase.driver.sql.query-processor]]. Moved here so it can live out its last
+  days in a place we don't have to look at it, and to discourage people from using it. Also convenient for seeing
+  everything that's deprecated at a glance.
+
+  Deprecated method impls should call [[log-deprecation-warning]] to gently nudge driver authors to stop using this
+  method."
+  (:require [clojure.tools.logging :as log]
+            [metabase.driver :as driver]
+            [metabase.query-processor.store :as qp.store]
+            [metabase.query-processor.util.add-alias-info :as add]
+            [metabase.util :as u]
+            [metabase.util.i18n :refer [trs]]))
+
+;; TODO -- this is actually pretty handy and I think we ought to use it for all the deprecated driver methods.
+(defn log-deprecation-warning
+  "Log a warning about usage of a deprecated method.
+
+    (log-deprecation-warning driver 'my.namespace/method \"v0.42.0\")"
+  [driver method-name deprecated-version]
+  (letfn [(thunk []
+            (log/warn
+             (u/colorize 'red
+               (trs "Warning: Driver {0} is using {1}. This method was deprecated in {2} and will be removed in a future release."
+                    driver method-name deprecated-version))))]
+    ;; only log each individual message once for the current QP store; by 'caching' the value with the key it is
+    ;; effectively memoized for the rest of the QP run for the current query. The goal here is to avoid blasting the
+    ;; logs with warnings about deprecated method calls, but still remind people regularly enough that it gets fixed
+    ;; sometime in the near future.
+    (if (qp.store/initialized?)
+      (qp.store/cached [driver method-name deprecated-version]
+        (thunk))
+      (thunk))))
+
+;;; DEPRECATED in 41
+
+(defmulti ^String field->alias
+  "Returns an escaped alias for a Field instance `field`.
+
+  DEPRECATED as of x.41. This method is no longer used by the SQL QP itself, but is still available for existing code
```

Comment:
And here?

Reply:
Edit: nevermind, I see what you did (these were already deprecated in a prior patch).